overlord/snapstate/backend/copydata.go

```diff
@@ -272,12 +272,10 @@ type UndoInfo struct {
 }
 
 // InitExposedSnapHome creates and initializes ~/Snap/<snapName> based on the
-// specified revision. Must be called after the snap has been migrated. If no
-// error occurred, returns a non-nil undoInfo so that the operation can be undone.
-// If an error occurred, an attempt is made to undo so no undoInfo is returned.
-func (b Backend) InitExposedSnapHome(snapName string, rev snap.Revision) (undoInfo *UndoInfo, err error) {
-	opts := &dirs.SnapDirOptions{HiddenSnapDataDir: true}
-
+// specified revision. If no error occurred, returns a non-nil undoInfo so that
+// the operation can be undone. If an error occurred, an attempt is made to undo
+// so no undoInfo is returned.
+func (b Backend) InitExposedSnapHome(snapName string, rev snap.Revision, opts *dirs.SnapDirOptions) (undoInfo *UndoInfo, err error) {
```

Comment:
It's used but there's no diff because there was already an `opts` variable (deleted from line 279 in the old version)

Reply:
I see, I read the diff too quickly... why are we passing this in now?

Reply:
Previously this would always be called after the migration to the hidden directory was done so we didn't need it. Now user maybe call the debug command with the with or without the hidden snap dir so now we need that information.